src/classes/RD2_EntryFormController.cls

```diff
@@ -354,29 +320,158 @@ public with sharing class RD2_EntryFormController {
+                PS_CommitmentRequest.RequestBody requestBody = getRequestBody(rd, paymentMethodToken);
+                
+                return processCommitment(rd.Id, rd.CommitmentId__c, JSON.serialize(requestBody));
 
-            return request.getRequestBody(recordId, paymentMethodToken);
+            } else {
+                return buildOkResponse();
+            }
         }
 
-        /***
-        * @description Sends an HttpRequest so that matching Recurring Commitment is created in Elevate
-        * @param jsonRequestBody JSON containing parameters for the commitment call request body
-        * @return UTIL_Http.Response
-        */
-        public UTIL_Http.Response sendRequest(Id recordId, String jsonRequestBody) {
+        private npe03__Recurring_Donation__c getOldRecurringDonation(Id recordId) {
+            npe03__Recurring_Donation__c oldRd = null;
+
+            if (recordId != null) {
+                List<npe03__Recurring_Donation__c> oldRds = new RD2_QueryService().getRecurringDonationsBase(new Set<Id>{ recordId });
+                if (oldRds != null && !oldRds.isEmpty()) {
+                    oldRd = oldRds[0];
+                }
+            }
+
+            return oldRd;
+        }
+
+        private npe03__Recurring_Donation__c constructRecurringDonation(Id recordId, String jsonRecord, npe03__Recurring_Donation__c oldRd) {
+            npe03__Recurring_Donation__c rd = (npe03__Recurring_Donation__c) JSON.deserialize(
+                jsonRecord, npe03__Recurring_Donation__c.class
+            );
+
+            if (recordId != null) {
+                rd.Id = recordId;
+            }
+
+            if (rd.Id == null && String.isBlank(rd.Status__c)) {
+                rd.Status__c = 'Active';//TODO populate defaults applied by DML (trigger), currency in a multi-currency org? etc.
+            }
+
+            List<npe03__Recurring_Donation__c> oldRds = new List<npe03__Recurring_Donation__c>();
+            if (oldRd != null) {
+                oldRds.add(oldRd);
+            }
+            
+            auditor.adjust(new List<npe03__Recurring_Donation__c>{ rd }, oldRds);
+
+            return rd;
+        }
+
+        private void validate(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd, String paymentMethodToken) {
+            List<npe03__Recurring_Donation__c> oldRds = new List<npe03__Recurring_Donation__c>();
+            if (oldRd != null) {
+                oldRds.add(oldRd);
+            }
+
+            List<RD2_ValidationService.Notification> notifications = auditor.validate(
+                new List<npe03__Recurring_Donation__c>{ rd }, oldRds
+            );
+
+            if (!notifications.isEmpty()) {
+                UTIL_AuraEnabledCommon.throwAuraHandledException(
+                    notifications[0].getErrors()[0]
+                );
+            }        
+        }
+
+        private Boolean shouldSendToElevate(npe03__Recurring_Donation__c rd, String paymentMethodToken) {
```

Comment:
RD that is closed, or for which credit card is not added/changed should not be processed by the commitment edit request (W-8585272).